avoidance/src/transform_buffer.cpp

```diff
@@ -0,0 +1,128 @@
+#include "avoidance/transform_buffer.h"
+
+namespace avoidance {
+
+TransformBuffer::TransformBuffer(float buffer_size_s) : buffer_size_(ros::Duration(buffer_size_s)) {
+  mutex_.reset(new std::mutex);
+};
+
+std::string TransformBuffer::get_key(const std::string& source_frame, const std::string& target_frame) const {
+  return source_frame + "_to_" + target_frame;
+}
+
+bool TransformBuffer::interpolateTransform(const tf::StampedTransform& tf_earlier, const tf::StampedTransform& tf_later,
+                                           tf::StampedTransform& transform) const {
+  // check if the requested timestamp lies between the two given transforms
+  if (transform.stamp_ > tf_later.stamp_ || transform.stamp_ < tf_earlier.stamp_) {
+    return false;
+  }
+
+  ros::Duration timeBetween = tf_later.stamp_ - tf_earlier.stamp_;
```

Comment:
they cannot be equal. We never insert two equal TFs into the buffer. the criterion for inserting a TF is that it must be newer than the most recent buffered one (see `insertTransform`)

Reply:
Since this is a protected function I'll take that. But generally, your code should catch these things and never assume that people use your functions correctly

Reply:
actually the check just a line above prevents that. It is checked that the requested timestamp lies between the two given transforms. That would fail if both given transforms have the same timestamp

Reply:
The above check verifies only the relationship between those two and transform.stamp_. 
So it's possible e.g.
transform.stamp = 1.0
later.stamp =  0.0
earlier.stamp = 0.0

And the check will pass and you divide by zero